### PR TITLE
PendingChannels() sets confirmation_height

### LINKED ISSUE #1217: confirmation_height is always zero
### Background

while a channel is in pending state lncli pendingchannels always shows "confirmation_height" : 0

### Your environment

lnd 0.41beta
 4.15.0-20-genericx86_64 GNU/Linux
bitcoind 0.16

### Steps to reproduce

open a channel to LND from other client
list channels with lncli pendingchannels
wait for some confirmations

### Expected behaviour

confirmation_height should show the number of confirmations

### Actual behaviour

confirmation_height always show 0

### PR DESCRIPTION
**Pull request by tvolk131**
Fixes #1217

Once a channel's funding transaction is included in a block, the `confirmation_height` field is set to the correct value.

Note: I attempted to add test coverage for these changes, but `rpcserver_test.go` is basically empty... If anyone can point towards a good place to add a test, I'm happy to do so.

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -3267,6 +3267,10 @@ func (r *rpcServer) fetchPendingOpenChannels() (pendingOpenChannels, error) {
 		utx := btcutil.NewTx(localCommitment.CommitTx)
 		commitBaseWeight := blockchain.GetTransactionWeight(utx)
 		commitWeight := commitBaseWeight + input.WitnessCommitmentTxWeight
+		confirmationHeight := uint32(0)
```
Comment (guggero) [inline rpcserver.go:3270]: nit: declare as `var confirmationHeight uint32`? And add empty line above to visually group this a bit.

File: rpcserver.go [hunk 2]
```diff
@@ -3286,9 +3287,9 @@ func (r *rpcServer) fetchPendingOpenChannels() (pendingOpenChannels, error) {
 				Private:              isPrivate(pendingChan),
 			},
 			ConfirmationHeight: confirmationHeight,
-			CommitWeight: commitWeight,
-			CommitFee:    int64(localCommitment.CommitFee),
-			FeePerKw:     int64(localCommitment.FeePerKw),
+			CommitWeight:       commitWeight,
```
Comment (guggero) [inline rpcserver.go:3290]: This commit should be squashed with the previous one.

**Review by guggero — COMMENTED** [review]
I haven't tested this, so I'm not 100% sure the short channel ID will ever be set on the pending channel while it's still pending. But I assume you tested this locally on regtest?

You're correct, the RPC server doesn't have many unit tests. The best place to test this is probably an integration test. For example `lntest/itest/lnd_funding_test.go`.

**Comment by tvolk131** [comment]
This turned out to be a formidable technical foe... Beyond my expertise, at least for now. Closing in favor of #6472.

**Pull request closed without merging**